lib/Finance/Quote/Bourso.pm

```diff
@@ -125,194 +123,146 @@ sub bourso {
-                unless ($opcvm) {
-                    $info{ $stocks, "success" } = 0;
-                    $info{ $stocks, "errormsg" } =
-                        "$stocks retrieval not supported.";
-                    next;
-                }
-
-                # the stock is a delayed OPCVM
-
-                my $infoelem =
-                    ( $tree->look_down( 'id', 'quote-infos-page' ) )[0];
-                $infoelem =
-                    ( $infoelem->look_down( 'class', 'q-details span-1-2' ) )
-                    [0];
-
-                my @rows = $infoelem->look_down( '_tag', 'tr' );
-                foreach my $i ( 0 .. $#rows ) {
-                    my $row = $rows[$i];
-                    unless ( $row->attr('class') ) {
-                        next;
-                    }
-
-                    my @cells     = $row->look_down( '_tag', 'td' );
-                    my $keytext   = ( $cells[0] )->as_text;
-                    my $valuetext = ( $cells[2] )->as_text;
-
-                    $tempinfo{$keytext} = $valuetext;
-                }
-            }
-            else {
-                # regular stock
-
-                my $infoelem;
-                my $quote_infos_page =
-                    ( $tree->look_down( 'id', 'quote-infos-page' ) )[0];
-                $infoelem = ( $quote_infos_page->look_down(
-                                                   'class', 'q-details span-1-2'
-                              )
-                )[0];
-                $infoelem = ( $quote_infos_page->look_down( 'class', 'bd' ) )[0]
-                    if ( !defined $infoelem );    # needed for warrants
-
-                my @rows = $infoelem->look_down( '_tag', 'tr' );
-                foreach my $i ( 0 .. $#rows ) {
-                    my $row   = $rows[$i];
-                    my @cells = $row->look_down( '_tag', 'td' );
-                    my $j     = 0;
-                    if ( $cells[0]->attr('rowspan') ) {
-                        $j = 1;
-                    }
-                    if ( $cells[0]->attr('colspan') ) {
-                        next;
-                    }
-
-                    my $keytext   = ( $cells[$j] )->as_text;
-                    my $valuetext = ( $cells[ $j + 1 ] )->as_text;
-
-                    $tempinfo{$keytext} = $valuetext;
-                }
-            }
+            # retrieve SYMBOL
+	    my $div = $tp->get_tag( 'div' );
+	    my($tag, $attr, $attrseq, $rawtxt) = @{ $div };
+	    my $symbol = $attr->{'data-ist'};
+	    if ($symbol) {
+		$info{ $stocks, "symbol" } = $symbol;
+
+		# print initial values
+		# print $tag."\t".$attr->{'data-ist-init'}."\n";
+		# remove surrounding braces
+		my $line = substr( $attr->{'data-ist-init'}, 1);
+		chop ($line);
+		# print $line."\n";
+		# retrieve data from csv line
+		my $csv = Text::CSV->new ({ escape_char => "\\", allow_loose_quotes => 1, binary => 1 , quote_char => '' })
```

Comment:
Content of `data-ist-init` is simple JSON, would be better to simply parse it with something like:

```perl
use JSON qw( decode_json );

# ...

my $json_data = JSON::decode_json $line;
$json_data->{'last'}
```

Reply:
Good point and thank you for doing the code review.
@jmpacquet - Could you please look into the changes suggested by Alexandre.
Thank you both once again.

Reply:
Boursorama site changed two days ago. As I had to, I modified the code and  followed Alexandre advice.